File: go/problem22.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	test1 := []string{`THIS`, `IS`, `ONLY`, `A`, `TEST`}
	fmt.Println(NameScores(test1))
}

func NameScores(input []string) int {

	sort.Strings(input)
	sum := 0
	for index, value := range input {
		scr := calculateScore(value)
		sum += (index + 1) * scr
	}
	return sum
}

func calculateScore(word string) int {

	score := 0
	chars := []rune(word)
	for i := 0; i < len(chars); i++ {
		score += int(chars[i]) - 64
	}
	return score
}

File: python/problem22/problem22.py
def nameScores(wordList):
    sum=0
    wordList.sort()
    for i in range(0, len(wordList)):
        sum += (i+1)*calculateScore(wordList[i])
    return sum

def calculateScore(inp):
    score = 0
    for i in range(0, len(inp)):
        score += ord(inp[i])-64
    return score

if __name__=='__main__':
    test1 = ['I', 'REPEAT', 'THIS', 'IS', 'ONLY', 'A', 'TEST'] #['THIS', 'IS', 'ONLY', 'A', 'TEST']
    print(nameScores(test1))
